### PR TITLE
SessionID as string

### PR DESCRIPTION
**Pull request by lbloder**
## :scroll: Description
Follow-up to #3818

## :bulb: Motivation and Context
Replace usage of `UUID` in `Session` with String and generate ID using `SentryUUID` class.

## :pencil: Checklist

- [ ] I reviewed the submitted code.
- [ ] I added tests to verify the changes.
- [ ] No new PII added or SDK only sends newly added PII if `sendDefaultPII` is enabled.
- [ ] I updated the docs if needed.
- [ ] Review from the native team if needed.
- [ ] No breaking change or entry added to the changelog.
- [ ] No breaking change for hybrid SDKs or communicated to hybrid SDKs.

### CODE REVIEW

File: sentry/src/main/java/io/sentry/Session.java [hunk 1]
```diff
@@ -79,7 +78,7 @@ public Session(
       final @Nullable Date timestamp,
       final int errorCount,
       final @Nullable String distinctId,
-      final @Nullable UUID sessionId,
```
Comment (adinauer) [inline sentry/src/main/java/io/sentry/Session.java:82]: We should write this down in the changelog as breaking change.

File: sentry/src/main/java/io/sentry/Session.java [hunk 2]
```diff
@@ -142,7 +141,7 @@ public boolean isTerminated() {
     return distinctId;
   }
 
-  public @Nullable UUID getSessionId() {
```
Comment (adinauer) [inline sentry/src/main/java/io/sentry/Session.java:145]: This should also be on the changelog as breaking change.

**Review by adinauer — APPROVED** [review]
LGTM, just needs changelog entry with breaking changes mentioned

File: sentry/src/main/java/io/sentry/Session.java [hunk 3]
```diff
@@ -451,12 +450,9 @@ public static final class Deserializer implements JsonDeserializer<Session> {
...
         switch (nextName) {
           case JsonKeys.SID:
-            String sidString = null;
-            try {
-              sidString = reader.nextStringOrNull();
-              sessionId = UUID.fromString(sidString);
-            } catch (IllegalArgumentException e) {
-              logger.log(SentryLevel.ERROR, "%s sid is not valid.", sidString);
+            String sid = reader.nextStringOrNull();
+            if (sid != null && sid.length() == 32) {
```
Comment (romtsn) [inline sentry/src/main/java/io/sentry/Session.java:454]: should we still keep the log call in the else branch?

File: sentry/src/main/java/io/sentry/Session.java [hunk 4]
```diff
@@ -79,7 +78,7 @@ public Session(
       final @Nullable Date timestamp,
       final int errorCount,
       final @Nullable String distinctId,
-      final @Nullable UUID sessionId,
+      final @Nullable String sessionId,
```
Comment (romtsn) [inline sentry/src/main/java/io/sentry/Session.java:81]: do we have to apply some validation still that this is compliant with the uuid format?

**Review by romtsn — APPROVED** [review]
couple of comments to address, but lgtm otherwise

**Pull request merged**